src/consul.rs

```diff
@@ -1,9 +1,20 @@
 use anyhow::Result;
 use reqwest::Url;
 use serde::{Deserialize, Serialize};
-use std::time::{Duration, Instant, SystemTime};
+use std::{
+    collections::HashMap,
+    time::{Duration, Instant, SystemTime},
+};
 use tokio::time::sleep;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DnsTag {
+    pub hostname: String,
+    pub type_: String,
```

Comment:
I think this should be an enum enumerating the specific types we currently want to support setting.  (`A`, `AAAA`, and `CNAME`.)